### PR TITLE
Classic tester cleanup alt

### PR DESCRIPTION
**Pull request by dt27182**
As per Chick's suggestion, we can instantiate a default dynamic variable inside the Builder object so that new Module() can be called out side of the Driver.elaborate function. This implementation avoids the problem of exposing a possibly confusing "elaborateModule" function. Jack and I think that this is probably the better alternative to [#183]([https://github.com/ucb-bar/chisel3/pull/183)

Would really like @sdtwigg and @aswaterman to comment on this.

### CODE REVIEW

File: chiselFrontend/src/main/scala/Chisel/internal/Builder.scala [hunk 1]
```diff
@@ -92,7 +92,7 @@ private[Chisel] class DynamicContext {
 
 private[Chisel] object Builder {
   // All global mutable state must be referenced via dynamicContextVar!!
-  private val dynamicContextVar = new DynamicVariable[Option[DynamicContext]](None)
+  private val dynamicContextVar = new DynamicVariable[Option[DynamicContext]](Some(new DynamicContext))
 
   def dynamicContext: DynamicContext =
```
Comment (sdtwigg) [inline chiselFrontend/src/main/scala/Chisel/internal/Builder.scala]: Can we make this private as well?

Builder should have accessors for all the 'global state' that route through dynamicContext.

File: src/main/scala/Chisel/Tester.scala [hunk 2]
```diff
@@ -1,39 +0,0 @@
-// See LICENSE for license details.
```
Comment (sdtwigg) [inline src/main/scala/Chisel/Tester.scala]: Er so, can we get a Readme update as to where this is going?

Is chisel-testers 'released' now? (at the very least, needs a LICENSE)

**Comment by sdtwigg** [comment]
It will take me a a little time to work through the DynamicVariable 'documentation' to verify that a default is OK here (and won't just violate the thread safety desires).

File: chiselFrontend/src/main/scala/Chisel/internal/Builder.scala (hunk 1 above)
Comment (dt27182) [inline chiselFrontend/src/main/scala/Chisel/internal/Builder.scala]: This seems to be a non trivial change. I think this should be raised as a separate issue.

File: src/main/scala/Chisel/Tester.scala (hunk 2 above)
Comment (dt27182) [inline src/main/scala/Chisel/Tester.scala]: We are releasing chisel-testers and I will add a LICENCE file to that repo. Not sure what needs to be done in the README for this.

**Comment by aswaterman** [comment]
I think this is probably fine

**Comment by jackkoenig** [comment]
I think it should be squashed into 2 commits, 1 deleting Tester.scala and 1 providing the default DynamicContext

**Comment by dt27182** [comment]
Actually after further testing, this fix doesn't work because the dynamic variable isn't refreshed when new Module() is called multiple times, leading to mangled names. I have modified the classic tester to use the lazy val trick to get an module instance for now. I have also made a separate pull request that just contains the removal of the Tester.scala file since that conflicts with the implementation in chisel-testers.

**Pull request closed without merging**